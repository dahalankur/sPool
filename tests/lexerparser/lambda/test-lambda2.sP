# lambdas with legal statement lists

lambda int (int x, int y):
    ;

lambda int (int x, int y):
    a = 5
    3.43
    c(1, false);

lambda string (bool x, int y):
    def int myFunc (int x):
        return x;
    ;
