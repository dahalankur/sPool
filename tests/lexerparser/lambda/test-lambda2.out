PROGRAM {[ 
    LAMBDA(INTTY, FORMALS((INTTY, x), (INTTY, y)), []) 
    LAMBDA(INTTY, FORMALS((INTTY, x), (INTTY, y)), [ 
        ASSIGN(a, INT(5)) 
        FLOAT(3.43) 
        CALL(c, ARGS(INT(1), BOOL(false)))]) 
    LAMBDA(STRINGTY, FORMALS((BOOLTY, x), (INTTY, y)), [ 
        FUN(NOSTORE, INTTY, myFunc, FORMALS((INTTY, x)), [ 
            RETURN(VAR(x))])])]}
